output.go

```diff
@@ -184,6 +186,57 @@ func (vmOutput *VMOutput) GetFirstReturnData(asType vm.ReturnDataKind) (interfac
 	return nil, fmt.Errorf("can't interpret return data")
 }
 
+// GetMaxOutputTransferIndex returns the maximum output transfer index
+func (vmOutput *VMOutput) GetNextAvailableOutputTransferIndex() uint32 {
+	maxTransferIndex := uint32(0)
+	noTransferFound := true
+	for _, account := range vmOutput.OutputAccounts {
+		for _, transfer := range account.OutputTransfers {
+			noTransferFound = false
+			if transfer.Index > maxTransferIndex {
+				maxTransferIndex = transfer.Index
+			}
+		}
+	}
+
+	if noTransferFound {
+		return 0
+	}
+
+	return maxTransferIndex + 1
+}
+
+// ReindexTransfers from VMOutput
+func (vmOutput *VMOutput) ReindexTransfers(nextIndexProvider NextOutputTransferIndexProvider) error {
+
+	if nextIndexProvider == nil {
+		return ErrNilTransferIndexer
+	}
+
+	firstAccountWithTransfers := true
+	for _, account := range vmOutput.OutputAccounts {
+		if len(account.OutputTransfers) != 0 {
+			if firstAccountWithTransfers {
+				firstAccountWithTransfers = false
+				continue
+			}
+
+			// this is not the first account with transfers and the others were not indexed
+			if nextIndexProvider.GetCrtTransferIndex() == 0 {
```

Comment:
you should check first whether the outPutTransfers have been indexed. If yes, then it is all good.